src/mlpack/methods/ann/layer/convolution_impl.hpp

```diff
@@ -502,13 +502,13 @@ void ConvolutionType<
   const size_t paddedRows = this->inputDimensions[0] + padWLeft + padWRight;
   const size_t paddedCols = this->inputDimensions[1] + padHTop + padHBottom;
 
-  arma::Cube<typename MatType::elem_type> inputTemp(
+  CubeType inputTemp(
       const_cast<MatType&>(usingPadding ? inputPadded : input).memptr(),
       paddedRows, paddedCols, inMaps * batchSize, false, false);
```

Comment:
We'll have to come back to this, but basically we'll have to replace these with some form of `MakeAlias()` or something like this, instead of directly calling the Armadillo advanced constructor.  But that can happen later. :+1: